src/main/java/tdl/client/queue/transport/RemoteBroker.java

```diff
@@ -1,75 +1,130 @@
 
-    public RemoteBroker(String hostname, int port, String uniqueId, int requestTimeoutMillis) throws JMSException {
-        String brokerURL = String.format("tcp://%s:%s", hostname, port);
-        ActiveMQConnectionFactory connectionFactory = new ActiveMQConnectionFactory(brokerURL);
-        connection = connectionFactory.createConnection();
+    public RemoteBroker(String hostname, int port, String uniqueId, int requestTimeoutMillis) throws JMSException, EventProcessingException {
+        System.out.println("     RemoteBroker creation [start]");
 
+        Config config = ConfigFactory.load();
+        config.checkValid(ConfigFactory.defaultReference());
+
+        serviceEndpoint = String.format("http://%s:%d", hostname, port);
+        client = createAWSClient(
+                serviceEndpoint,
+                config.getString("sqs.signingRegion"),
+                config.getString("sqs.accessKey"),
+                config.getString("sqs.secretKey")
+        );
+
+//        String brokerURL = String.format("tcp://%s:%s", hostname, port);
+//        ActiveMQConnectionFactory connectionFactory = new ActiveMQConnectionFactory(brokerURL);
+//        connection = connectionFactory.createConnection();
+//
         LoggerFactory.getLogger(RemoteBroker.class).debug("Connecting to the remote broker");
-        connection.start();
-        session = connection.createSession(false, Session.CLIENT_ACKNOWLEDGE);
+//        connection.start();
+//        session = connection.createSession(false, Session.CLIENT_ACKNOWLEDGE);
+//
+        String requestQueue = uniqueId + "-req";
+        messageConsumer = new SqsEventQueue(client, serviceEndpoint, requestQueue);
 
-        String requestQueue = uniqueId + ".req";
-        messageConsumer = session.createConsumer(session.createQueue(requestQueue));
+        String responseQueue = uniqueId + "-resp";
+        messageProducer = new SqsEventQueue(client, serviceEndpoint, responseQueue);
 
-        String responseQueue = uniqueId + ".resp";
-        messageProducer = session.createProducer(session.createQueue(responseQueue));
-        messageProducer.setDeliveryMode(DeliveryMode.NON_PERSISTENT);
+        queueEventHandlers = new QueueEventHandlers();
+        queueEventHandlers.put(ExecuteCommandEvent.class, capturedEvents::add);
 
-        this.requestTimeoutMillis = requestTimeoutMillis;
-        serializationProvider = new JsonRpcSerializationProvider();
+        System.out.println("     RemoteBroker creation [end]");
     }
 
-    public Optional<Request> receive() throws BrokerCommunicationException {
+    public List<Request> receive() throws BrokerCommunicationException {
+        System.out.println("     RemoteBroker receive [start]");
         try {
-            //Debt: We should have no timeout. This method could exit if we put a special close message in the queue
-            StringMessage messageText = new StringMessage(messageConsumer.receive(requestTimeoutMillis));
```

Comment:
I had a proper look at this and I thing that we should not use the `SqsEventQueue` but use the SDK directly.
The `SqsEventQueue` is designed for high-throughput asynchronous message processing, which makes it suitable for the participant event queue but not for this purpose.

This method should issue an AWS ReceiveMessageRequest 
```
import com.amazonaws.services.sqs.model.ReceiveMessageRequest;
...
receiveMessageRequest = new ReceiveMessageRequest();
receiveMessageRequest.setMaxNumberOfMessages(1);
receiveMessageRequest.setQueueUrl(queueUrl);
receiveMessageRequest.setWaitTimeSeconds(toSeconds(requestTimeoutMillis, or(20)));
...
List<Message> messages = client.receiveMessage(receiveMessageRequest).getMessages();
```

Reply:
Resolved via 885f9145681f6560b78f931bdc687a4c6146fef6